Add general help page (#2356)
YunoHost/doc: pages/07.help/help.md

---
title: Help me
template: docs
taxonomy:
    category: docs
routes:
  default: '/help-me'
---

Even though YunoHost contributors try to continuously perfect it, YunoHost itself or this documentation are not free from bugs or flaw. To let us help you, however, you need to come prepared.

# Dos

## Be courteous and patient
Help in the chatrooms or on the forum is given by our users and contributors, all volunteers. You may be in a hurry, you may be exhausted by fruitless trials... Have a break, have a sip of your preferred soothing beverage, then continue reading. We need you in a good mindset to help you.

## Explain the context
Your request needs to be stated in a complete but concise way. Explain plainly what you want to achieve and in what manner it is not working for you.

## Show errors and logs
This is absolutely paramount if YunoHost or a command fails. It always explains why, through a single error code or a full wall of seemingly illegible text.

For the latter, YunoHost provides green "Share with YunoPaste" buttons when something fails or in the Services page of the webadmin. Click on it, it will open a new web page, and you can then share its URL.

> ![YunoPaste button](image://yunopaste.png)

Author: ```suggestion
![YunoPaste button](image://yunopaste.png)

```

![YunoPaste from install error](image://yunopaste_install.mp4?loop=1&controls=0&autoplay=1&muted)
![YunoPaste from service](image://yunopaste_service.mp4?loop=1&controls=0&autoplay=1&muted)

If a page fails to load properly, learn to investigate with your browser console (opened with the F12 key): its Network tab can show you HTTP error codes (including the infamous 403, 404, 500...), and its Console can output errors too.

Screenshots are most welcome.

To guide you, on the forum you need to fill in the Support request template.

# Don'ts

## Avoid the infamous [XY Problem](https://xyproblem.info/)

> In a nutshell, explain the root cause of your issue. You may share your attempted solution, but why you need that potential solution is more important.

Reviewer: ```suggestion
In a nutshell, explain the root cause of your issue. You may share your attempted solution, but *why* you need that potential solution is more important.
```

## Do not be vague
Do not assume we understand what you are talking about, do not take shortcuts, do not rephrase, do not simplify, do not trim the error messages.

> Do not write "it does not work", "it does not want", "it does not load", ["computer says no"](https://en.wikipedia.org/wiki/Computer_says_no).

Reviewer: ```suggestion
Do not write "it does not work", "it does not want", "it does not load", ["computer says no"](https://en.wikipedia.org/wiki/Computer_says_no).

Explain what is wrong, and how it is displayed onscreen. There is, for example, an important difference between a completely blank page and a blank page with a minimalistic error code.
```

> Do not ask your question like a poll, as in "does anyone else has...". Most likely very few people have the same issue or need as you, and even less read the forum or the chatrooms.

Reviewer: ```suggestion
## Ask your question directly

No need to ask if you can ask. It's a support chat/forum, therefore yes : you can ask your question.

Do not ask your question like a poll, as in "Did anyone managed to get app XY working ?". If it ain't working for you, then explain directly what issue you got, explain the context, share the logs. 
```

> Explain what is wrong, and how it is displayed onscreen. There is, for example, an important difference between a completely blank page and a blank page with a minimalistic error code.

Reviewer: ```suggestion
```

(regrouped right after the "Do not write 'it does not work'" etc)

---
title: Help me
template: docs
taxonomy:
    category: docs
routes:
  default: '/help-me'
---

Even though YunoHost contributors try to continuously perfect it, YunoHost itself or this documentation are not free from bugs or flaw. To let us help you, however, you need to follow these guidelines.

Support is only available on [the forum](https://forum.yunohost.org?target=_blank) or our [chatrooms](/chat_rooms?target=_blank).
We are also on Mastodon, but we do not offer help on this platform.

> # Dos

Reviewer: ```suggestion
# Do's
```

Reviewer: :+1:

> To guide you, on the forum you need to fill in the Support request template.

Author: ```suggestion
To allow us to help you on the forum, you absolutely need to fill in the Support request template.
```

## Show errors and logs
This is absolutely paramount if YunoHost or a command fails. It always explains why, through a single error code or a full wall of seemingly illegible text.

For the latter, YunoHost provides green "Share with YunoPaste" buttons when something fails or in the Services page of the webadmin. Click on it, it will open a new web page, and you can then share its URL.

![YunoPaste button](image://yunopaste.png)

![YunoPaste from install error](image://yunopaste_install.mp4?loop=1&controls=0&autoplay=1&muted)
![YunoPaste from service](image://yunopaste_service.mp4?loop=1&controls=0&autoplay=1&muted)

If your issue is not related to an installation or upgrade process but instead is about an app failing load properly, ideally you can investigate with your browser console (opened with the F12 key): its Network tab can show you HTTP error codes (including the infamous 403, 404, 500...), and its Console can output errors too.

Screenshots are most welcome.

To guide you, on the forum you need to fill in the Support request template.

> # Don'ts

Reviewer: ```suggestion
# Dont's
```

Author: I respectfully disagree: https://dictionary.cambridge.org/dictionary/english/dos-and-don-ts 😛

Reviewer: Ooooh alrighty sorry haha

## Avoid the infamous [XY Problem](https://xyproblem.info/)

> In a nutshell, explain the root cause of your issue. You may share your attempted solution, but *why* you need that potential solution is more important.

Reviewer: ```suggestion
In a nutshell, explain the root cause of your issue. You may share your attempted solution, but explaining ***why*** you need that potential solution is more important.
```
